src/ShaderTools/Hlsl/Binding/Binder.Declarations.cs

```diff
@@ -267,7 +267,7 @@ private BoundNode BindFunctionDefinition(FunctionDefinitionSyntax declaration, S
                 Bind(declaration.Semantic, BindVariableQualifier);
 
             var functionBinder = (functionOwner != null && functionOwner.Kind == SymbolKind.Class)
```

Comment:
There's some decision logic based on that, which Glyph to choose, what keyword to write (struct/class).